tests/test_jemx_plugin.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+import logging
+import requests
+import json
+
+from tests.test_basic import ask
```

Comment:
does this actually work for you? It should not since tests is not a package. You need to add a __init__.py in it.
It's a bit unconventional though, but there are many ways.

Reply:
yes it does, but yeah absolutely is not the best

Reply:
but did you add `__init__.py`?

Reply:
I actually did not

Reply:
I realized it is worse than just not the best. `tests` directory exists in many plugins and packages. So importing will sometimes not work.
What is even worse yet, loop_ask currently exists in tests/test_basic.. in several packages, causing confusion.
I think these common functions should be just put in dispatcher,  https://github.com/oda-hub/dispatcher-app/blob/master/cdci_data_analysis/pytest_fixtures.py or similar. In fact, they can be made into fixtures.

Reply:
Ok to move those in a common place within the dispatcher, but declared as normal functions, not as fixtures/. Am I correct?

Reply:
as you like. But it should not be in the normal dispatcher code, it's testing code. Could be in pytest_fixtures.py .
But they can be also fixtures, why not. pytest support special functionality for fixtures